### DISCUSSION TITLE
What can I combine with useComboBox? Are useGridList and useTable allowed?

### DISCUSSION THREAD

**Discussion by donaldpipowitch** [adobe/react-spectrum#3916] [Q&A]
Sometimes it is a bit hard to wrap my head around all the hooks react-arias offers. So I hope I don't confuse things here.

I want users to select options from a list. Because the list should be filterable with a text input I use `useComboBox` instead of `useSelect`. Right now it looks like `useComboBox` _has_ to be paired with `useListBox` which handles the popup list box. At least this is how I understand the documentation.

However `useGridList` has some features that `useListBox` doesn't have. When I have a look at the documentation (e.g. _"If your list items contain only static content such as text and images, then consider using useListBox instead for a slightly better screen reader experience (especially on mobile)."_) I'd consider them as interchangeable - wherever I can use `useListBox` I should also be able to use `useGridList`.
It actually looks like I can also use the `listBoxProps` returned from `useComboBox` as the first parameter of `useGridList`. At least TypeScript doesn't complain, because the props for ListBox and GridLast have the same base type. Is this usage allowed?

Going a step further. In case my GridList should look like a Table: Should I rather visually fake a Table with `useGridList` or can I actually combine `useComboBox` with `useTable` in a correct way?

**Edit:** Related discussion, but in the scope of a Select, not a ComboBox: https://github.com/adobe/react-spectrum/discussions/3896

**Comment by donaldpipowitch** [comment]
Okay, I tried to look a little bit into the source code. I think you _cannot_ properly use `useGridList` with `useComboBox`. One of the missing features is `shouldUseVirtualFocus` if I see correctly. This is not implemented for `useGridList`. One would have to switch the real focus between the GridList and the text input from ComboBox. 😕

**Reply by snowystinger** [reply]
I think it would be hard to have a GridList inside a Combobox. Specifically because ArrowKeys would end up in conflict. Left and Right arrow keys in a Combobox should go left and right in the input so you can edit the text there. But left and right in a grid would navigate cells/columns.
